charts/clamav/templates/clamav-deployment.yaml

```diff
@@ -0,0 +1,129 @@
+        - containerPort: 8080
+          name: api
+          protocol: TCP
+        livenessProbe:
+          httpGet:
+            path: /v1alpha/healthz
+            port: 8080
+          initialDelaySeconds: 10
+          timeoutSeconds: 5
+          periodSeconds: 5
+          failureThreshold: 10
+        readinessProbe:
+          httpGet:
+            path: /v1alpha/healthz
+            port: 8080
+          initialDelaySeconds: 10
+          timeoutSeconds: 5
+          periodSeconds: 5
+          failureThreshold: 10
+        resources: {{ toYaml .Values.clamav.resources | nindent 10 }}
+      - name: nginx
+        image: {{.Values.nginxProxy.image}}:{{.Values.nginxProxy.version}}
+        env:
+        - name: PROXY_SERVICE_HOST
+          value: 127.0.0.1
+        - name: PROXY_SERVICE_PORT
+          value: "8080"
+        - name: ENABLE_UUID_PARAM
+          value: "FALSE"
+        - name: NAXSI_USE_DEFAULT_RULES
+          value: "FALSE"
+        - name: ERROR_REDIRECT_CODES
+          value: "599"
+        - name: CLIENT_MAX_BODY_SIZE
+          value: '{{.Values.clamav.limits.fileSize}}'
+        - name: SERVER_CERT
+          value: /certs/tls.crt
+        - name: SERVER_KEY
+          value: /certs/tls.key
+        - name: SSL_CIPHERS
+          value: ECDHE-RSA-AES128-GCM-SHA256:AES256+EECDH:AES256+EDH:!aNULL
+        - name: ADD_NGINX_SERVER_CFG
+          value: |
+            add_header Cache-Control private;
+            add_header X-Forwarded-Proto "https";
+        livenessProbe:
+          httpGet:
+            path: /
+            port: 8080
+          initialDelaySeconds: 10
+          timeoutSeconds: 5
+          periodSeconds: 5
+          failureThreshold: 10
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 8080
+          initialDelaySeconds: 10
+          timeoutSeconds: 5
+          periodSeconds: 5
+          failureThreshold: 10
+        ports:
+         - name: https
+           containerPort: 10443
+        resources: {{ toYaml .Values.nginxProxy.resources | nindent 10 }}
+        volumeMounts:
+          - mountPath: /certs
+            name: certs
+      initContainers:
+      - name: signatures
+        image: "{{ .Values.clamav.image }}:{{ default .Chart.AppVersion .Values.clamav.version }}"
+        # --no-dns prevents failures when upstream mirrots are out of sync and allows us
+        # to use locally tested signatures rather than untested upstream signatures
+        command: ['freshclam', '--stdout', '--no-dns']
+        resources:
+          requests:
+            memory: 1000M
+          limits:
+            memory: 1000M
+        volumeMounts:
+          - mountPath: /var/lib/clamav
+            name: signatures
+          - mountPath: /etc/clamav
+            name: config-volume
+      securityContext:
+        runAsNonRoot: true
+      volumes:
+      - name: certs
+        secret:
+          secretName: {{ include "clamav.certsecretname" . }}
```

Comment:
If you don't create a cert, this secret never gets created.

Reply:
The goal here is enabling bring your own certificate rather than no secret at all

Reply:
In that case I think you should mention that if you choose not to create a cert, you need to make this manually in the `values.yaml` file